user_gen.py

```diff
@@ -0,0 +1,178 @@
+import numpy as np
+
+
+def simp_poisson_process(rate=1, size=None):
+    sum_exp_draws = np.cumsum(np.random.exponential(scale=rate, size=size))
+    # sum_exp_draws = [sum(exp_draws[:i]) for i, x in enumerate(exp_draws)]
+    return sum_exp_draws
+
+
+def poisson_plus(lam=1, plus=1, size=None):
```

Comment:
It's either that or pass it as a `lambda` function to `func`. I tend to disprefer lambda functions and so I try to avoid them for default values pretty much anywhere.